staging/src/k8s.io/apiserver/pkg/apis/apiserver/v1alpha1/types.go

```diff
@@ -167,3 +167,52 @@ type TracingConfiguration struct {
 	// Defaults to 0.
 	SamplingRatePerMillion *int32 `json:"samplingRatePerMillion,omitempty" protobuf:"varint,2,opt,name=samplingRatePerMillion"`
 }
+
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// PriorityAndFairnessConfiguration holds API Priority and Fairness parameters.
+type PriorityAndFairnessConfiguration struct {
+	metav1.TypeMeta `json:",inline"`
+
+	// WorkEstimator holds work estimator parameters.
+	// +optional
+	WorkEstimator WorkEstimatorConfiguration `json:"workEstimator,omitempty"`
+}
+
+// WorkEstimatorConfiguration holds work estimator parameters.
+type WorkEstimatorConfiguration struct {
+	// ListWorkEstimator holds work estimator parameters related to list requests.
+	// +optional
+	ListWorkEstimator ListWorkEstimatorConfiguration `json:"listRequests,omitempty"`
+	// MutatingWorkEstimator holds work estimator parameters
+	// related to requests that mutate watched objects.
+	MutatingWorkEstimator MutatingWorkEstimatorConfiguration `json:"mutatingRequests,omitempty"`
+
+	// MaximumSeats is the maximum number of seats a request can occupy
+	// +optional
+	MaximumSeats *uint64 `json:"maximumSeats,omitempty"`
+}
+
+// ListWorkEstimatorConfiguration holds work estimator parameters related to list requests.
+type ListWorkEstimatorConfiguration struct {
+	// ObjectsPerSeat is a number of objects from a single list request per seat.
+	// +optional
+	ObjectsPerSeat *float64 `json:"objectsPerSeat,omitempty"`
+}
+
+// MutatingWorkEstimatorConfiguration holds work estimator
+// parameters related to watches of mutating objects.
+type MutatingWorkEstimatorConfiguration struct {
+	// EventAdditionalDuration is the additional amount of time that a mutating request occupies seats
+	// to account for the work involved in sending the corresponding WATCH notifications. Setting its
+	// value to 0 effectively disables this behavior but a request may still occupy seats.
```

Comment:
(1) "but a request may still occupy seats" is vague, it could mean "but a request still occupies a seat while it executes".

(2) I think "but a request may still occupy seats" is trying to say that the extended behavior to account for WATCH notifications is _not_ fully disabled (i.e., the larger amount of seats is occupied while the request executes even though the extra seats are subsequently used for zero time).  So saying "0 effectively disables this behavior" denies an important detail.

We should either correctly describe the current behavior OR change the current behavior so that 0 additional duration fully disables the extra behavior AND clearly document this here.

My vote is for the latter, as I think it is a fairly natural meaning for zero here and relieves the need to define an unnatural meaning for 0 WatchesPerSeat.